pulumi/provider.go

```diff
@@ -0,0 +1,36 @@
+package pulumi
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Provider -
+func Provider() *schema.Provider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("PULUMI_ACCESS_TOKEN", nil),
+				Description: "The access token for Pulumi's cloud",
+			},
+		},
+		ResourcesMap: map[string]*schema.Resource{},
+		DataSourcesMap: map[string]*schema.Resource{
+			"pulumi_stack_outputs": dataSourceStackOutputs(),
+		},
+		ConfigureContextFunc: providerConfigure,
+	}
+}
+
+func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
```

Comment:
```suggestion
func providerConfigure(ctx context.Context, d *schema.ResourceData) (string, diag.Diagnostics) {
```

Reply:
I just tried this, but `ConfigureContextFunc` needs this to be an interface I guess

Reply:
Odd, considering you're returning a string which should satisfy the interface{} requirements? I guess I'm forgetting how Go types work in these cases